go-controller/pkg/ovn/service.go

```diff
@@ -165,6 +167,130 @@ func (ovn *Controller) syncServices(services []interface{}) {
 	}
 }
 
+func (ovn *Controller) createService(service *kapi.Service) error {
+	klog.V(5).Infof("Creating service %s", service.Name)
+	if !util.IsClusterIPSet(service) {
+		return fmt.Errorf("no cluster IP for service %s found", service.Name)
+	} else if len(service.Spec.Ports) == 0 {
+		return fmt.Errorf("no Ports specified for service")
+	}
+
+	for _, svcPort := range service.Spec.Ports {
+		var port int32
+		if util.ServiceTypeHasNodePort(service) {
+			port = svcPort.NodePort
+		} else {
+			port = svcPort.Port
+		}
+		if port == 0 {
+			continue
+		}
+
+		protocol := svcPort.Protocol
+		if protocol == "" || (protocol != TCP && protocol != UDP) {
```

Comment:
`kapi.ProtocolTCP`/`kapi.ProtocolUDP` instead of `TCP`/`UDP`?

Reply:
Here and in `deleteService`, it's a nit, but seeing as how you changed it elsewhere.

Reply:
yeah iirc I think I left it here because I was going to have to change a lot more. I'll go back and look at it again and see if I can change it. Ideally I would prefer using just the kapi one.

Reply:
The create function uses the constants in service.go as well. I'm just going to make another PR after this and clean this up everywhere.

Reply:
@trozet if protocol is `SCTP`, why are we forcing it to `TCP`? Shouldn't we just error out here?